letsencrypt/plugins/standalone/authenticator.py

```diff
@@ -301,11 +301,14 @@ def already_listening(self, port):  # pylint: disable=no-self-use
         :param int port: The TCP port in question.
         :returns: True or False."""
 
-        listeners = [conn.pid for conn in psutil.net_connections()
-                     if conn.status == 'LISTEN' and
-                     conn.type == socket.SOCK_STREAM and
-                     conn.laddr[1] == port]
         try:
```

Comment:
[PEP8](https://www.python.org/dev/peps/pep-0008/): "Additionally, for all try/except clauses, limit the try clause to the absolute minimum amount of code necessary. Again, this avoids masking bugs."

This clause has grown too much now. Please refactor something like:

```python
try:
    net_connections = psutil.net_connections()
except psutil.AccessDenied as error:
    logger.info("Access denied when trying to list network connections: "
                "%s. Are you root?", error)
    # this function is just a pre-check that often causes false positives
    # and problems in testing (c.f. #680 on Mac, #255 generally); we will
    # fail later in bind() anyway
    return False

try:
    listeners = [... for conn in net_connections ...]
...
```

In fact everything else apart from `psutil.Process(pid).name()` should be taken out of the `try/except` block, but this is out of scope for this PR.